presto-spi/src/main/java/com/facebook/presto/spi/plan/InputDistribution.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.plan;
+
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.MINIMAL_CLASS, property = "@type")
```

Comment:
what does this do?

Reply:
It serializes the subtype ids using a minimal class name with the @type json key, to match up with what we have in the presto_protocol_core.yml and presto_protocol.yml files.  Or at least that is my understanding of it.  I'm not a Jackson expert.